packages/backend-core/src/cache/appMetadata.ts

```diff
@@ -61,11 +61,8 @@ export async function getAppMetadata(appId: string) {
     }
     await client.store(appId, metadata, expiry)
   }
-  // we've stored in the cache an object to tell us that it is currently invalid
-  if (isInvalid(metadata)) {
-    throw { status: 404, message: "No app metadata found" }
-  }
-  return metadata as App
+
+  return metadata as App & { state: AppState }
```

Comment:
I think `state` here is a bit confusing, in that it can only have a single value. 
The naming is a bit confusing as well because what does "invalid" actually mean. 

I prefer how stripe sdk handles this, where the function would either return `App | DeletedApp`. Where the former would look like:

```ts
interface DeletedApp { 
   appId: string,
   /**
    * Always true for a deleted object
    */
   deleted: true
}
```
and the latter
```ts
interface App extends Document {
   // ...other props
   deleted?: boolean
}
```

Reply:
If we were to keep an implementation closer to the current model, I think it should be more like:
```ts
interface DeletedApp {
   state: AppState
}

export async function getAppMetadata(appId: string): Promise<App | DeletedApp> {
   ...
   if (isInvalid(metadata)) {
      return { state: AppState.INVALID }
   }
   ...
   return metadata
}
```

Preference would be on a deleted flag because it's a lot more clear what's going in. In either implementation I don't think we should return `App & { state: AppState }` as that's not what we return, it'll be one or the other.

Reply:
Good suggestion - we wouldn't need to cast it then with `as`. Will update to a dedicated `interface` and union type